lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -274,9 +334,112 @@ def test_some_NaNs_in_nhbooded_cube(self):
         result = self.plugin.process(nbhooded_cube)
         self.assertArrayAlmostEqual(expected_result, result.data)
 
-    # def test_some_NaNs_in_nhbooded_cube(self):
-        # """landsea mask"""
+    def test_no_weights_given(self):
+        """No weights are given so a mean without weighting should be used to
+           collapse the coordinate."""
+
+        nbhood_data = np.array([[[0, 0, 0., 0, 0],
+                                 [0.6, 0.3, 0, 0, 0],
+                                 [0, 0, 0, 0, 0],
+                                 [0, 0, 0, 0, 0],
+                                 [0, 0, 0, 0, 0]],
+                                [[0.3, 0.3, 1, 1, 1],
+                                 [0, 0, 1, 1, 1],
+                                 [0.6, 0.9, 1, 1, 1],
+                                 [1, 1, 1, 0.9, 0.6],
+                                 [1, 1, 1, 0, 0]],
+                                [[0, 0, 0, 0, 0],
+                                 [0, 0, 0, 0, 0],
+                                 [0, 0, 0, 0, 0],
+                                 [0, 0, 0, 0, 0],
+                                 [0, 0, 0, 0.3, 0.9]]])
+
+        expected_result = np.array([[0.1, 0.1, 0.333333, 0.333333, 0.333333],
+                                    [0.2, 0.1, 0.333333, 0.333333, 0.333333],
+                                    [0.2, 0.3, 0.333333, 0.333333, 0.333333],
+                                    [0.333333, 0.333333, 0.333333, 0.3, 0.2],
+                                    [0.333333, 0.333333, 0.333333, 0.1, 0.3]])
+        nbhooded_cube = self.weights_cube.copy(nbhood_data)
+        plugin = CollapseMaskedNeighbourhoodCoordinate("topographic_zone")
+        result = plugin.process(nbhooded_cube)
+        self.assertArrayAlmostEqual(expected_result, result.data)
 
+    def test_landsea_mask_in_weights(self):
+        """Test that the final result from collapsing the nbhood retains the
+           mask from weights input."""
+        nbhood_data = np.ones((3, 5, 5))
+        nbhood_data[0] = nbhood_data[0]*0.1
+        nbhood_data[1] = nbhood_data[1]*0.2
+        nbhood_data[2] = nbhood_data[2]*0.1
+        nbhood_data[0, 0:2, 0] = np.nan
+        nbhood_data[2, 2:4, 4] = np.nan
+        nbhooded_cube = self.weights_cube.copy(nbhood_data)
+        mask = np.array([[[1, 1, 1., 0, 0],
+                          [1, 1, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 1, 1, 1, 1]],
+                         [[1, 1, 1., 0, 0],
+                          [1, 1, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 1, 1, 1, 1]],
+                         [[1, 1, 1., 0, 0],
+                          [1, 1, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 1, 1, 1, 1]]])
```

Comment:
Done, and factored it out where its used twice in the same test class twice, but I haven't moved the third case because its in a different test class. I can do more refactoring shared cubes and stuff if you want. It would probably make this file less lines and maybe easier to follow.